### DISCUSSION TITLE
Timeout Error - Sense Nov 2022 ?

### DISCUSSION THREAD

**Discussion by jsebright** [ptarmiganlabs/butler-sheet-icons#261] [Q&A] [answered]
Hi,  am trying to generate some files on a windows server that I've just upgraded to Nov 2022 release. Have got the commandline working enough so that it processes all the sheets, but after that I am getting a timeout error.

2022-12-13T09:34:44.939Z error: QSEoW APP: TimeoutError: Waiting for selector `.//*[@id="hub-sidebar"]/div[1]/div[1]/div/div/div` failed: Waiting failed: 30000ms exceeded

I get the same error on another server where I've managed to get it to run before, but have recently upgraded to Nov 2022 on that as well.
Have Qlik changed something on the hub page?

**Comment by mountaindude** [comment]
Good question.

I just upgraded to 2022-Nov the other week and haven't tested BSI against that QSEoW version yet.
Will do so and report back.

It's quite possible that Qlik can change the client code. 
It's in fact quite possible they *will* do so at some point...
To be honest I'm a bit surprised BSI has worked as long as it has without any adoptions whatsoever to new Sense versions.
This might be it though - will check.

**Comment by mountaindude** [comment]
@jsebright Indeed, I am seeing the same thing in Qlik Sense 2022-Nov patch 2.

I don't yet know what's causing this but will investigate.
Work relating to this bug is tracked in #264.

**Comment by mountaindude** [comment, marked as answer]
@jsebright 
Version 3.0.0 was just released, available here: https://github.com/ptarmiganlabs/butler-sheet-icons/releases/tag/butler-sheet-icons-v3.0.0

It adds a new, mandatory `--sense-version` command line parameter. 
Valid values are currently `pre-2022-Nov` and `2022-Nov`.

So... get the 3.0.0 version of BSI, then try running it again, adding `--sense-version 2022-Nov` to the command that failed previously.
Should work 🤓

**Reply by jsebright** [reply]
That's done the trick - worked as expected without errors. Many thanks for this.
Fingers crossed that the codes you're looking for don't change too frequently.
